api/apiclient_test.go

```diff
@@ -790,7 +790,9 @@ func (s *apiclientSuite) TestOpenTimesOutOnLogin(c *gc.C) {
 }
 
 func (s *apiclientSuite) TestOpenTimeoutAffectsDial(c *gc.C) {
+	sync := make(chan struct{})
 	fakeDialer := func(ctx context.Context, urlStr string, tlsConfig *tls.Config, ipAddr string) (jsoncodec.JSONConn, error) {
+		close(sync)
 		<-ctx.Done()
```

Comment:
do we want a bare channel read here? or do we want to wrap it in a time.After(testing.LongWait)

Reply:
We know that the context is killed within the test, so it progresses.